### PR TITLE
Add routing for the projector microservice

### PR DESCRIPTION
**Pull request by james-hinton**

### CODE REVIEW

**Comment by ivica3730k** [comment]
checking now

File: app/main/controller/transformer_controller.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        if "file" not in request.files:
+            return {"message": "No file part in the request"}, HTTP_STATUS_BAD_REQUEST
+        shapefile = request.files["file"]
+
+        if shapefile.filename == "":
+            return {
+                "message": "No file selected for uploading"
+            }, HTTP_STATUS_BAD_REQUEST
+
+        if not allowed_file(shapefile.filename):
```
Comment (ivica3730k) [inline app/main/controller/transformer_controller.py:41]: what happens if someone accidentally (or on purpose) puts a big file in?

Comment (ivica3730k) [reply app/main/controller/transformer_controller.py:41]: Can we limit it on the front-end at least to be only zip-s smaller than n size?

File: app/main/controller/transformer_controller.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+from flask import request
+from flask_restx import Resource
+from werkzeug.utils import secure_filename
+from typing import Tuple
+
+from ..aad.auth_decorators import AuthDecorator
+from ..service.transformer_service import send_file_to_transformer
+from ..util.dto import TransformerDto
+
+ALLOWED_EXTENSIONS = {"zip"}
```
Comment (ivica3730k) [inline app/main/controller/transformer_controller.py:10]: could prepend _ to these as they should be private?

File: app/main/controller/transformer_controller.py (hunk 1 above)
Comment (james-hinton) [reply app/main/controller/transformer_controller.py:41]: i've got a feeling there's a better place for that, maybe some flask has some env to limit payload max size ? as by the time it gets to the controller it's already sent

Comment (ivica3730k) [reply app/main/controller/transformer_controller.py:41]: https://blog.miguelgrinberg.com/post/handling-file-uploads-with-flask

**Review by ivica3730k — COMMENTED** [review]
Can we not use something like this:
https://pypi.org/project/os-grid-reference/

**Comment by james-hinton** [comment]
Cool I prefer this one to the one I was using, good find.

**Comment by ivica3730k** [comment]
That one will do 27700 output tho, is that worth just converting with something else then?

**Comment by ivica3730k** [comment]
fixed up conflict just now in init py

**Comment by ivica3730k** [comment]
Did you limit upload to only .zip files?

**Pull request merged**